runtime/src/iree/hal/drivers/vulkan/debug_reporter.cc

```diff
@@ -6,18 +6,23 @@
 
 #include "iree/hal/drivers/vulkan/debug_reporter.h"
 
+#include <atomic>
 #include <cstddef>
 #include <cstdio>
 
+#include "iree/base/assert.h"
 #include "iree/hal/drivers/vulkan/status_util.h"
 
 struct iree_hal_vulkan_debug_reporter_t {
   iree_allocator_t host_allocator;
   VkInstance instance;
   iree::hal::vulkan::DynamicSymbols* syms;
   int32_t min_verbosity;
+  bool check_errors;
   const VkAllocationCallbacks* allocation_callbacks;
   VkDebugUtilsMessengerEXT messenger;
+
+  std::atomic<bool> did_error{false};
```

Comment:
prefer iree_atomic_* for atomics - don't want to add more c++ we have to remove later